activemq-client/src/main/java/org/apache/activemq/transport/tcp/QualityOfServiceUtils.java

```diff
@@ -125,15 +126,17 @@ private static int adjustDSCPForECN(int dscp)
         // The only way to see if there are any values set for the ECN is to
         // read the traffic class automatically set by the system and isolate
         // the ECN bits.
-        Socket socket = new Socket();
-        try {
+        try (Socket socket = new Socket()) {
             int systemTrafficClass = socket.getTrafficClass();
             // The 1st and 2nd bits of the system traffic class are the ECN
             // bits.
             return (dscp << 2) | (systemTrafficClass & 3);
         } catch (SocketException e) {
             throw new IllegalArgumentException("Setting Differentiated Services"
                 + " not supported: " + e);
+        } catch (IOException e) {
```

Comment:
Why not returning IAE here ?

Reply:
I don't think IAE makes sense, there should be a different exception as it's not an illegal argument so it's a misleading exception to throw.

If the goal is to simply make the exception unchecked then the best thing to do is just wrap it in [UncheckedIOException](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/io/UncheckedIOException.html)

Reply:
done